lib/generators/manageiq/provider/provider_generator.rb

```diff
@@ -25,23 +29,51 @@ def create_provider_files
 
     alias provider_name file_name
 
+    def self.manager_types
+      @manager_types ||= {
+        "automation"    => "AutomationManager",
+        "cloud"         => "CloudManager",
+        "configuration" => "ConfigurationManager",
+        "container"     => "ContainerManager",
+        "infra"         => "InfraManager",
+        "monitoring"    => "MonitoringManager",
+        "network"       => "NetworkManager",
+        "physical"      => "PhysicalInfraManager",
+        "provisioning"  => "ProvisioningManager",
+        "storage"       => "StorageManager"
+      }
+    end
+
+    def validate_manager_type!
+      return unless manager_type.nil?
+      raise "Invalid manager_type: #{options[:manager_type]}\nMust be one of #{self.class.manager_types.keys.join(", ")}"
+    end
+
+    def manager_type
+      @manager_type ||= self.class.manager_types[options[:manager_type]]
+    end
+
+    def manager_path
+      manager_type.underscore
+    end
+
     def create_dummy
-      template "app/models/%plugin_path%/cloud_manager/event_catcher/runner.rb"
-      template "app/models/%plugin_path%/cloud_manager/event_catcher/stream.rb"
-      template "app/models/%plugin_path%/cloud_manager/metrics_collector_worker/runner.rb"
-      template "app/models/%plugin_path%/cloud_manager/refresh_worker/runner.rb"
-      template "app/models/%plugin_path%/cloud_manager/event_catcher.rb"
-      template "app/models/%plugin_path%/cloud_manager/metrics_capture.rb"
-      template "app/models/%plugin_path%/cloud_manager/metrics_collector_worker.rb"
-      template "app/models/%plugin_path%/cloud_manager/refresh_worker.rb"
-      template "app/models/%plugin_path%/cloud_manager/refresher.rb"
-      template "app/models/%plugin_path%/cloud_manager/vm.rb"
-      template "app/models/%plugin_path%/inventory/collector/cloud_manager.rb"
-      template "app/models/%plugin_path%/inventory/parser/cloud_manager.rb"
+      template "app/models/%plugin_path%/%manager_path%/event_catcher/runner.rb"
+      template "app/models/%plugin_path%/%manager_path%/event_catcher/stream.rb"
```

Comment:
I think it'd be cool to allow `--event-catcher` or `--no-event-catcher` directives also since most initial providers don't do events or metrics at first, and I've seen a few examples of generated providers where the metrics collector is just left unchanged because they don't understand what it is.

Reply:
Should events and metrics be separated generators maybe?  Like, someone starts with an inventory generator, then follows up with events or metrics?  Then, if given as an option during the inventory generator, it adds it in?

Reply:
Yeah I like that